swarm/storage/hasherstore.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package storage
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/ethereum/go-ethereum/crypto/sha3"
+	"github.com/ethereum/go-ethereum/swarm/storage/encryption"
+)
+
+type chunkEncryption struct {
+	spanEncryption encryption.Encryption
+	dataEncryption encryption.Encryption
+}
+
+type hasherStore struct {
+	store           ChunkStore
+	hashFunc        SwarmHasher
+	chunkEncryption *chunkEncryption
+}
+
+func newChunkEncryption() *chunkEncryption {
+	return &chunkEncryption{
+		spanEncryption: encryption.New(0, math.MaxUint32, sha3.NewKeccak256),
+		dataEncryption: encryption.New(4096, 0, sha3.NewKeccak256),
+	}
+}
+
+func NewHasherStore(chunkStore ChunkStore, hashFunc SwarmHasher, toEncrypt bool) *hasherStore {
```

Comment:
comment exported functions

Reply:
Fixed in 8016f18cf13758bab198eff526a1b3726ab256d6